src/plot.js

```diff
@@ -410,21 +410,36 @@ function inferChannelScales(channels) {
   }
 }
 
-function addScaleChannels(channelsByScale, stateByMark, filter = yes) {
+function addScaleChannels(channelsByScale, stateByMark, options, filter = yes) {
   for (const {channels} of stateByMark.values()) {
     for (const name in channels) {
       const channel = channels[name];
       const {scale} = channel;
       if (scale != null && filter(scale)) {
-        const scaleChannels = channelsByScale.get(scale);
-        if (scaleChannels !== undefined) scaleChannels.push(channel);
-        else channelsByScale.set(scale, [channel]);
+        // Geo marks affect the default x and y domains if there is no
+        // projection. Skip this (as an optimization) when a projection is
+        // specified, or when the domains for x and y are specified.
+        if (scale === "projection") {
+          if (!hasProjection(options)) {
+            const [x, y] = getGeometryChannels(channel);
+            if (options.x?.domain === undefined) addScaleChannel(channelsByScale, "x", x);
+            if (options.y?.domain === undefined) addScaleChannel(channelsByScale, "y", y);
```

Comment:
Ideally we skip getGeometryChannels if both options.x?.domain and options.y?.domain are defined.

Reply:
yes we should do this